src/Microsoft.SqlTools.ServiceLayer/BatchParser/BatchParserWrapper.cs

```diff
@@ -0,0 +1,415 @@
+            {
+                positions.Add(new System.Tuple<int/*startOffset*/, int/*length*/>(offsets[batchIndex], lengths[batchIndex]));
+            }
+
+            List<string> batchTextList = GetBatchText(sqlScript, positions);
+           
+            List<BatchDefinition> batchDefinitions = new List<BatchDefinition>();
+            int endLine;
+            int endColumn;
+            System.Tuple<int, int> endLineColumn;
+
+            for (int i = 0; i < offsets.Count; i++)
+            {   
+                StringReader reader = new StringReader(sqlScript);
+
+                // if it's the last batch
+                if (i == offsets.Count - 1)
+                {      
+                    endLineColumn = GetLastEndLineColumn(reader);
+                    endLine = endLineColumn.Item1;
+                    endColumn = endLineColumn.Item2;
+                }
+                else {
+                    endLine = startLineColumns[i+1].Item1-1;
+                    endColumn = GetEndColumn(reader, endLine);
+                }
+
+                // create a batch definition for each batch
+                BatchDefinition batchDefinition = new BatchDefinition(
+                    batchTextList[i], //batchText
+                    startLineColumns[i].Item1 + 1, //startLine
+                    endLine, //endLine
+                    startLineColumns[i].Item2 + 1, //startColumn
+                    endColumn); //endColumn
+
+                batchDefinitions.Add(batchDefinition);           
+            }
+         
+            startLineColumns = null;
+            lengths = null;
+           
+            return batchDefinitions;
+        }
+
+        /// <summary>
+        /// Helper method used to return the individual text for the batches
+        /// </summary>
+        /// <param name="sqlScript"></param>
+        /// <param name="batchResult"></param>
+        /// <returns></returns>
+        private static List<string> GetBatchText(string sqlScript,
+            List<System.Tuple<int /*startOffset*/, int /*length*/>> batchResult)
+        {
+            List<string /*batchText*/> batchTextList = new List<string>();
+            foreach (var result in batchResult)
+            {
+                int startOffset = result.Item1;
+                int length = result.Item2;
+                batchTextList.Add(sqlScript.Substring(startOffset, length));
+            }
+            return batchTextList;
+        }
+
+        /// <summary>
+        /// Helper method used to Convert line/column information in a file to offset
+        /// </summary>
+        /// <param name="content"></param>
+        /// <param name="positions"></param>
+        /// <returns></returns>
+        private static IList<int> ConvertToOffsetSortedInput(string content, IList<System.Tuple<int, int>> positions)
+        {
+            IList<int/*offset*/> offsetList = new List<int>();
+
+            if (!string.IsNullOrEmpty(content) && (positions.Count > 0))
```

Comment:
This method makes no sense based on current design - the positions list it's passed is always empty (it's the start columns one). I think this points to use needing to review & test this class more thoroughly. Let's sync offline on this.